command-derive/command-derive-impl/src/lib.rs

```diff
@@ -71,92 +61,204 @@ pub fn command_derive(input: TokenStream) -> TokenStream {
+                let field_num = unnamed.unnamed.len();
+                let flatten_flags = unnamed.unnamed.into_iter().map(|field| {
+                    field
+                        .attrs
+                        .into_iter()
+                        .filter_map(|attr| attr.parse_meta().ok())
+                        .any(|meta| CommandMarker("flatten").validate(&meta).is_ok())
+                });
+                let named_fields: Vec<_> =
+                    (0..field_num).map(|i| format_ident!("x{}", i)).collect();
+                let (named_fields_to_str_piece, read_fields_piece): (Vec<_>, Vec<_>) = (0..field_num)
+                    .zip(flatten_flags)
+                    .map(|(i, flatten)| {
+                        let fld_ident = format_ident!("x{}", i);
+                        match flatten {
+                            false => (
+                                quote!{ res.push(#fld_ident.to_string()) },
+                                quote!{ args.next().ok_or_else(|| ::anyhow::anyhow!("Not enough args"))?.as_ref().parse().map_err(|e| ::anyhow::anyhow!("{}", e))? }
+                            ),
+                            true => (
+                                quote!{ res.extend(#fld_ident.into_iter()) },
+                                quote!{ ::command_derive::FromStrIter::from_str_iter(&mut args)? }
+                            )
+                        }
+                    })
+                    .unzip();
+                let idx_fields_piece: Vec<_> =
+                    (0..field_num).map(|i| Member::Unnamed(i.into())).collect();
+                let pattern = quote! {
+                    #ident (#(#named_fields,)*)
+                };
+                (
+                    named_fields_to_str_piece,
+                    read_fields_piece,
+                    idx_fields_piece,
+                    pattern,
+                )
+            }
+            Fields::Unit => (Vec::new(), Vec::new(), Vec::new(), quote! { #ident }),
         };
         var_idents.push(ident);
         codes.push(code);
-        patterns.push(pattern);
-        named_fields.push(named_fields_piece);
+        named_fields_to_str.push(named_fields_to_str_piece);
+        read_fields.push(read_fields_piece);
         idx_fields.push(idx_fields_piece);
+        patterns.push(pattern);
     }
-    (quote!{
-impl ::command_derive::Command for #enum_ident {
-    fn ident(&self) -> &'static str {
-        match self {
-            #(
-                #enum_ident::#var_idents { .. } => #codes,
-            )*
-                _ => panic!("`ident()` on skipped variant")
+    (quote! {
+    impl ::command_derive::Command for #enum_ident {
+        fn ident(&self) -> &'static str {
+            match self {
+                #(
+                    #enum_ident::#var_idents { .. } => #codes,
+                )*
+            }
+        }
+
+        fn extract_args(&self) -> Vec<String> {
+            use ::core::iter::IntoIterator;
```

Comment:
what is the point of using `::` at the start everywhere?

Reply:
This forces compiler to search for `core` in the root of the project. For example, if you write `use my_lib_core as core;`, then use this macro, it still will import "real" core internally